File: Python/Aulas/tap/treino_2/2291.cpp
#include <iostream>

using namespace std;

long long tabela[1000001];
long long n;

long long crivo(int n)
{
	int i = 1;
	while (i * i <= n)
	{
		tabela[i * i] += i;
		
		for(int j = i * i + i; j <= n; j += i)
		{
			tabela[j] += i + int(j/i);
		}
		i++;
	}
	
	for (int i = 1; i <= n; i++)
	{
		tabela[i] += tabela[i - 1];
	}
		
}

int main()
{
	crivo(1000000);
	cin >> n;
	while (n > 0)
	{
		cout << tabela[n] << endl;
		cin >> n;
	}
	return 0;
}

File: Python/Aulas/tap/treino_2/2291.py
def crivo(n):
    tabela = [0]*(n + 1)
    kDivisores = []
    
    R = int(n ** 0.5)
    for i in range(1, R + 1):
        tabela[i * i] += i
        
        for j in range(i*i + i, n + 1, i):
            tabela[j] += i + j//i

    for i in range(1, n + 1):
        tabela[i] += tabela[i - 1]

    return tabela

def main():
    divinos = crivo(1000000)

    n = int(input())
    while n > 0:
        print(divinos[n])
        n = int(input())

main()
